src/nvim/os/os_win_console.c

```diff
@@ -40,3 +40,13 @@ void os_replace_stdout_and_stderr_to_conout(void)
   const int conerr_fd = _open_osfhandle((intptr_t)conout_handle, 0);
   assert(conerr_fd == STDERR_FILENO);
 }
+
+bool os_has_vti(void)
+{
+  HANDLE handle = (HANDLE)_get_osfhandle(input_global_fd());
+  DWORD dwMode;
+  if (handle != INVALID_HANDLE_VALUE && GetConsoleMode(handle, &dwMode)) {
+    return dwMode & ENABLE_VIRTUAL_TERMINAL_INPUT;
```

Comment:
this might be a narrowing conversion?

```suggestion
    return !!(dwMode & ENABLE_VIRTUAL_TERMINAL_INPUT);
```